src/main/java/net/minecraftforge/common/data/ForgeLootTableProvider.java

```diff
@@ -0,0 +1,178 @@
+                found = true;
+            } else if (lootCondition instanceof Inverted && findAndReplaceInInverted((Inverted) lootCondition, from, to)) {
+                found = true;
+            }
+        }
+
+        return found;
+    }
+
+    private boolean findAndReplaceInParentedLootEntry(ParentedLootEntry entry, Item from, Tag<Item> to) {
+        LootEntry[] lootEntries = getField(ParentedLootEntry.class, entry, 0);
+        boolean found = false;
+
+        for (LootEntry lootEntry : lootEntries) {
+            if (findAndReplaceInLootEntry(lootEntry, from, to)) {
+                found = true;
+            }
+        }
+
+        return found;
+    }
+
+    private boolean findAndReplaceInLootEntry(LootEntry entry, Item from, Tag<Item> to) {
+        ILootCondition[] lootConditions = getField(LootEntry.class, entry, 0);
+        boolean found = false;
+
+        for (int i = 0; i < lootConditions.length; i++) {
+            if (lootConditions[i] instanceof Alternative && findAndReplaceInAlternative((Alternative) lootConditions[i], from, to)) {
+                found = true;
+            } else if (lootConditions[i] instanceof MatchTool && checkMatchTool((MatchTool) lootConditions[i], from)) {
+                lootConditions[i] = MatchTool.builder(ItemPredicate.Builder.create().tag(to)).build();
+                found = true;
+            }
+        }
+
+        return found;
+    }
+
+    private boolean findAndReplaceInAlternative(Alternative alternative, Item from, Tag<Item> to) {
+        ILootCondition[] lootConditions = getField(Alternative.class, alternative, 0);
+        boolean found = false;
+
+        for (int i = 0; i < lootConditions.length; i++) {
+            if (lootConditions[i] instanceof MatchTool && checkMatchTool((MatchTool) lootConditions[i], from)) {
+                lootConditions[i] = MatchTool.builder(ItemPredicate.Builder.create().tag(to)).build();
+                found = true;
+            }
+        }
+
+        return found;
+    }
+
+    private boolean findAndReplaceInInverted(Inverted inverted, Item from, Tag<Item> to) {
+        ILootCondition lootCondition = getField(Inverted.class, inverted, 0);
+
+        if (lootCondition instanceof MatchTool && checkMatchTool((MatchTool) lootCondition, from)) {
+            setField(Inverted.class, inverted, 0, MatchTool.builder(ItemPredicate.Builder.create().tag(to)).build());
+            return true;
+        } else if (lootCondition instanceof Alternative && findAndReplaceInAlternative((Alternative) lootCondition, from, to)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    private boolean checkMatchTool(MatchTool lootCondition, Item expected) {
+        ItemPredicate predicate = getField(MatchTool.class, lootCondition, 0);
+        Item item = getField(ItemPredicate.class, predicate, 4);
+        return item != null && item.equals(expected);
+    }
+
+    private <T, R> void setField(Class<T> clz, T inst, int index, R value) {
```

Comment:
Using access transformers might be better here, but involves removing final keyword

Reply:
I dont think it will be better, if that changes will be visible for modders

Reply:
probably better to add a getter, and then replace the whole Inverted instance instead of just the term inside.

Reply:
Removed setField method as proposed